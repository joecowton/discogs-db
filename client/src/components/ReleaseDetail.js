import _ from 'lodash';
import React, { Component } from 'react';
import ReactPlayer from 'react-player';
import { Link } from 'react-router-dom';
import { fetchRelease } from '../actions';
import { connect } from 'react-redux';
import Payments from './tools/Payments';

class ReleaseDetail extends Component {
  componentDidMount() {
    const { id } = this.props.match.params;
    this.props.fetchRelease(id);
  }

  mapImages(release) {
    return _.map(release.images, image => {
      return <image
        key={image.uri}
        src={image.uri}
        height="200" width="200" padding="10"
        onClick={() => console.log("here")}></image>
    });
  }

  mapArtists(release){return _.map(release.artists, artist => {
      return <div>{artist.name} {artist.join}</div>
    })
  }

  mapVideos(release){return _.map(release.videos, video => {
      return (
        <div className="container" style={{ display: 'inline-grid'}}>
          <div className="row">
            <div className="col-3">
              <ReactPlayer url={video.uri} height="100" width="100%" />
            </div>
          </div>
         </div>
      )
    })
  }

  renderRelease() {
    const { data } = this.props
      return (
        <li key={data.id} className="list-group-item" >
          <h1>{this.mapArtists(data)}</h1>
          <h2>{data.title}</h2>
          <p>{this.mapImages(data)}</p>
          {this.mapVideos(data)}
          <p><Payments release={data}/></p>
        </li>
      )
  }

  render() {
    const { data } = this.props;

    if (!data){
      return (
        <div className="progress white">
          <div className="indeterminate green"></div>
        </div>
      )
    }

    return (
      <div>
        <Link to="/releases/all">Back to Index</Link>
        <ul className="list-group center">
          {this.renderRelease()}
        </ul>
      </div>
    )
  }
}

function mapStateToProps({ data }, ownProps) {
  return { data: data[ownProps.match.params.id] }
}

export default connect(mapStateToProps, { fetchRelease } )(ReleaseDetail);
