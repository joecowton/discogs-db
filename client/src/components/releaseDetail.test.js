import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import ReleaseDetail from './ReleaseDetail';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('<ReleaseDetail />', () => {
    it('should render the component', () => {
        const initialState = { data: { id: 2 } };
        const store = mockStore(initialState);
        const props = { key: 'releaseDetail', match: { params: { id: 2 } } };
        const wrapper = shallow(<ReleaseDetail {...props} store={store} />);
        expect(wrapper).toMatchSnapshot();
    });
});
