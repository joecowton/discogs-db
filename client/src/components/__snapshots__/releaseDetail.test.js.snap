// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseDetail should render the component 1`] = `
<div>
  <Link
    replace={false}
    to="/releases/all"
  >
    Back to Index
  </Link>
  <ul
    className="list-group center"
  >
    <li
      className="list-group-item"
      key="2"
    >
      <h1 />
      <h2 />
      <p />
      <p>
        <Connect(Payments)
          release={
            Object {
              "artist": "",
              "id": 2,
              "label": "",
              "thumb": "",
              "title": "",
            }
          }
        />
      </p>
    </li>
  </ul>
</div>
`;
